Files/Dialogs/ConfirmDeleteDialog.xaml.cs

```diff
@@ -16,14 +18,26 @@ public enum MyResult
             Nothing
         }
 
-        public ConfirmDeleteDialog(bool deleteFromRecycleBin = false)
+        public ConfirmDeleteDialog(bool deleteFromRecycleBin, StorageDeleteOption deleteOption)
         {
             this.InitializeComponent();
 
             this.Result = MyResult.Nothing; //clear the result in case the value is set from last time
+            this.PermanentlyDelete = deleteOption;
 
             // If deleting from recycle bin disable "permanently delete" option
-            this.chkPermanentlyDelete.IsEnabled = !deleteFromRecycleBin;
+            this.chkPermanentlyDelete.IsEnabled = !deleteFromRecycleBin;            
+
+            this.Unloaded += ConfirmDeleteDialog_Unloaded;
+        }
+
+        private void ConfirmDeleteDialog_Unloaded(object sender, RoutedEventArgs e)
+        {
+            // x:Bind "forgets" to unregister bindings when unloading
+            System.Reflection.FieldInfo field = typeof(ConfirmDeleteDialog).GetField("Bindings", 
+                System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+            dynamic bindings = field.GetValue(this);
+            bindings.StopTracking();
```

Comment:
I've never done something like this before, so I'm curious why it's necessary in this case.

Reply:
I noticed the issue by putting a breakpoint [here](https://github.com/files-community/files-uwp/blob/c16cb7a728f7e3d5fa8fea9227c39aefd84c6dce/Files/View%20Models/InteractionViewModel.cs#L38) and seeing it was getting called more and more times when canceling and reopening the dialog.
This change is probably now unnecessary as I'm now binding `PermanentlyDelete` to a dialog local property instead of a static app property.
I guess it's worth checking if a similar issue exists for other Pages/Dialogs as it could be the source of memory leaks?